pkg/v1/tkg/test/tkgctl/shared/mhc_cc.go

```diff
@@ -0,0 +1,182 @@
+		options := framework.CreateClusterOptions{
+			ClusterName: clusterName,
+			Namespace:   namespace,
+			Plan:        input.Plan,
+			CniType:     input.Cni,
+		}
+
+		if input.E2EConfig.InfrastructureName == "vsphere" {
+			if endpointIP, ok := os.LookupEnv("CLUSTER_ENDPOINT_2"); ok {
+				options.VsphereControlPlaneEndpoint = endpointIP
+			}
+		}
+
+		clusterConfigFile, err := framework.GetTempClusterConfigFile(input.E2EConfig.TkgClusterConfigPath, &options)
+		Expect(err).To(BeNil())
+
+		defer os.Remove(clusterConfigFile)
+		err = tkgCtlClient.CreateCluster(tkgctl.CreateClusterOptions{
+			ClusterConfigFile: clusterConfigFile,
+			Edition:           "tkg",
+		})
+		Expect(err).To(BeNil())
+
+		// Create a new client to prevent it from reusing in memory configs of the old client
+		tkgCtlClient2, _ = tkgctl.New(tkgctl.Options{
+			ConfigDir: input.E2EConfig.TkgConfigDir,
+			LogOptions: tkgctl.LoggingOptions{
+				File:      filepath.Join(logsDir, clusterName+".log"),
+				Verbosity: input.E2EConfig.TkgCliLogLevel,
+			},
+		})
+
+		By(fmt.Sprintf("Generating credentials for workload cluster %q", clusterName))
+		err = tkgCtlClient2.GetCredentials(tkgctl.GetWorkloadClusterCredentialsOptions{
+			ClusterName: clusterName,
+			Namespace:   namespace,
+			ExportFile:  "",
+		})
+		Expect(err).To(BeNil())
+
+		By(fmt.Sprintf("Waiting for workload cluster %q nodes to be up and running", clusterName))
+		contextName := clusterName + "-admin@" + clusterName
+		framework.WaitForNodes(framework.NewClusterProxy(clusterName, "", contextName), 2)
+
+		By(fmt.Sprintf("Generating credentials for workload cluster %q", clusterName))
+		err = tkgCtlClient2.GetCredentials(tkgctl.GetWorkloadClusterCredentialsOptions{
+			ClusterName: clusterName,
+			Namespace:   namespace,
+		})
+		Expect(err).To(BeNil())
+
+		wcContextName := clusterName + "-admin@" + clusterName
+		wcProxy = framework.NewClusterProxy(clusterName, "", wcContextName)
+	})
+
+	It("mhc should remediate unhealthy machine", func() {
+		// Validate MHC
+		By(fmt.Sprintf("Getting MHC for cluster %q", clusterName))
+		mhcList, err := tkgCtlClient2.GetMachineHealthCheck(tkgctl.GetMachineHealthCheckOptions{
+			ClusterName: clusterName,
+			Namespace:   namespace,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(len(mhcList)).To(Equal(2))
+		mhc := mhcList[0]
+		Expect(mhc.Spec.ClusterName).To(Equal(clusterName))
+		Expect(len(mhc.Spec.UnhealthyConditions)).To(Equal(2)) // nolint:gomnd
+
+		// Delete MHC and verify if MHC is deleted
+		By(fmt.Sprintf("Deleting MHC for cluster %q", clusterName))
+		if tkgCtlClient2 == nil {
+			_, _ = GinkgoWriter.Write([]byte("tkgCtlClient is nil"))
+		}
+		err = tkgCtlClient2.DeleteMachineHealthCheck(tkgctl.DeleteMachineHealthCheckOptions{
+			ClusterName:            clusterName,
+			MachinehealthCheckName: mhc.Name,
+			Namespace:              namespace,
+			SkipPrompt:             true,
+		})
+		Expect(err).To(HaveOccurred())
```

Comment:
We should also add a check to make sure we are getting the error message here by comparing the output with the error message that we are expecting.
```
Expect(err.Error()).To(ContainSubstring("deleting machine health checks on clusterclass based clusters is not supported"))
```

Reply:
done